Extension/src/LanguageServer/ui.ts

```diff
@@ -438,9 +460,13 @@ export class UI {
     }
 }
 
-export function getUI(): UI {
+export async function getUI(): Promise<UI> {
     if (!ui) {
-        ui = new UI();
+        const experimentationService: IExperimentationService | undefined = await telemetry.getExperimentationService();
+        if (experimentationService !== undefined) {
+            const useNewUI: boolean | undefined = experimentationService.getTreatmentVariable<boolean>("vscode", "splitUIUsers");
+            ui = useNewUI ? new NewUI() : new OldUI();
```

Comment:
Could we also check the C_Cpp.experimentalFeatures setting to enable users to opt-in to using this? Otherwise, it makes testing harder and users can't opt-in.